cmd/run.go

```diff
@@ -97,45 +104,45 @@ func NewRunCommand(run *KoolRun) (runCmd *cobra.Command) {
 	// after a non-flag arg, stop parsing flags
 	runCmd.Flags().SetInterspersed(false)
 
-	// look for kool.yml on current working directory
-	_ = run.parser.AddLookupPath(run.envStorage.Get("PWD"))
-	// look for kool.yml on kool folder within user home directory
-	_ = run.parser.AddLookupPath(path.Join(run.envStorage.Get("HOME"), "kool"))
-
-	var (
-		usageTempl string
-		err        error
-	)
-
-	if usageTempl, err = getRunUsageTemplate(run, runCmd); err == nil {
-		runCmd.SetUsageTemplate(usageTempl)
-	} else if run.envStorage.IsTrue("KOOL_VERBOSE") {
-		run.Println("$ got an error trying to add available scripts to command usage template; error:", err.Error())
-	}
-
 	return
 }
 
-func getRunUsageTemplate(run *KoolRun, cmd *cobra.Command) (templ string, err error) {
-	var (
-		sb      strings.Builder
-		scripts []string
-	)
-
-	if scripts, err = run.parser.ParseAvailableScripts(); err != nil {
-		return
-	}
+// SetRunUsageFunc overrides usage function
+func SetRunUsageFunc(run *KoolRun, runCmd *cobra.Command) {
+	originalUsageText := runCmd.UsageString()
+	runCmd.SetUsageFunc(getRunUsageFunc(run, originalUsageText))
+}
 
-	sb.WriteString(cmd.UsageTemplate())
-	sb.WriteString("\n")
-	sb.WriteString("Available Scripts:\n")
+func getRunUsageFunc(run *KoolRun, originalUsageText string) func(*cobra.Command) error {
+	return func(cmd *cobra.Command) (err error) {
+		var (
+			sb      strings.Builder
+			scripts []string
+		)
+
+		// look for kool.yml on current working directory
+		_ = run.parser.AddLookupPath(run.envStorage.Get("PWD"))
+		// look for kool.yml on kool folder within user home directory
+		_ = run.parser.AddLookupPath(path.Join(run.envStorage.Get("HOME"), "kool"))
+
+		if scripts, err = run.parser.ParseAvailableScripts(); err != nil {
+			if run.envStorage.IsTrue("KOOL_VERBOSE") {
+				run.Println("$ got an error trying to add available scripts to command usage template; error:", err.Error())
+			}
+			return
+		}
 
-	for _, script := range scripts {
-		sb.WriteString("  ")
-		sb.WriteString(script)
+		sb.WriteString(originalUsageText)
```

Comment:
do we really need to have the `originalUseageText` coming from `runCmd.UsageString()` in the init() time, instead of calling it only here within this closure? So we would only generate it when actually needed... unless setting the function would generate an infinite loop, in which case it must be like it is right now.

Reply:
@fabriciojs yep, I tried calling `UsageString()` inside the closure and it caused an infinite loop, indeed. And unfortunately, the functions to process the default template are not exported. We could replicate them on our code, but any changes on cobra code could break it.

Reply:
OK! we can keep as is!